### PR TITLE
Feature/cp 3452/deleting note

### PR DESCRIPTION
**Pull request by nf2000**
Deleting a note 

https://chillibean.atlassian.net/browse/CP-3454

### CODE REVIEW

File: nims-notes-things/src/components/notes.js [hunk 1]
```diff
@@ -40,6 +55,13 @@ function Notes() {
             <p>{note.date}</p>
           </Header>
           <Content>{note.note}</Content>
+          <NavLink to={`/deleteNote/${note.id}`}></NavLink>
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/notes.js]: Do you need to wrap the delete button around the nav link?

File: nims-notes-things/src/redux/reducers/noteReducer.js [hunk 2]
```diff
@@ -15,6 +15,33 @@ export default function (state = initialState, action) {
     case actions.SET_SORTED:
       const setSorted = state.sorted;
       return { ...state, sorted: !setSorted };
+    case actions.SET_DELETED:
+      const ArrayOfNote = state.value.filter((note) => {
+        if (note.id === action.id) {
```
Comment (damonspencerroberts) [inline nims-notes-things/src/redux/reducers/noteReducer.js]:
```
if (note.id === action.id) {
          return false;
        }
        return true;
```
can turn into
```
return note.id !== action.id
```

File: nims-notes-things/src/components/Form.js [hunk 3]
```diff
@@ -95,11 +144,13 @@ const Form = (props) => {
             value={formData.note}
           />
         </InnerForm>
-      </StyledDiv>
-      <Button  type="submit" to="/notes">
-        Submit
-      </Button>
-    </StyledForm>
+        {!noteId ? (
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: switch these:
`{noteId ? Update : Submit}`

Comment (damonspencerroberts) [reply nims-notes-things/src/components/Form.js]: easier to read

**Review by damonspencerroberts — CHANGES_REQUESTED** [review]
Couple of comment but coming along well

File: nims-notes-things/src/components/notes.js (hunk 1 above)
Comment (nf2000) [reply nims-notes-things/src/components/notes.js:58]: i don't need the navlink there anymore -> removed it :)

**Review by damonspencerroberts — APPROVED** [review]

**Pull request merged**